sopel/plugins/rules.py

```diff
@@ -1068,6 +1104,22 @@ def is_global_rate_limited(self):
         rate_limit = datetime.timedelta(seconds=self._global_rate_limit)
         return (now - last_usage_at) <= rate_limit
 
+    def get_user_rate_message(self, nick):
+        if self._user_rate_message is not None:
+            return self._user_rate_message.format(nick=nick)
+        return self._default_rate_message
+
+    def get_channel_rate_message(self, nick, channel):
+        if self._channel_rate_message is not None:
+            return self._channel_rate_message.format(
+                nick=nick,
+                channel=channel,
+            )
+        return self._default_rate_message
+
+    def get_global_rate_message(self):
+        return self._global_rate_message or self._default_rate_message
```

Comment:
Since there's already some support for placeholders, what do you think of adding one for the time remaining until the rate limit expires? It's the main reason that _I_ wished for this feature to be built in, all those years ago.

Regardless of the above, these placeholders don't appear to be documented anywhere.

Reply:
Yup I thought about it. I wanted to put the bulk of the code first for review, it would be easy to add anything after that.

Reply:
I also just noticed that the global and default versions don't support even a `nick` placeholder. 🤔